### ISSUE TITLE
snickerbockers/jit_branch: WashingtonDC crashes when booting KallistiOS with the jit enabled

### ISSUE THREAD

**Issue by snickerbockers** [washingtondc-emu/washingtondc#33]
I noticed this while I was trying to run sh4div_test.

```
Total elapsed time: 2 seconds and 986315887 nanoseconds
26008304 SH4 CPU cycles executed
Performance is 8.709160 MHz (4.354580%)
ERROR: unable to continue due to unimplemented functionality
[sh4_reg_mmucr] = 0
[sh4_reg_tea] = 0
[sh4_reg_ttb] = 0
[sh4_reg_ptea] = 0
[sh4_reg_ptel] = 0
[sh4_reg_pteh] = 0
[sh4_reg_qacr1] = 0
[sh4_reg_qacr0] = 0
[sh4_reg_ccr] = 929
[sh4_reg_r15] = 8c0b2e9c
[sh4_reg_r14] = 4328
[sh4_reg_r13] = 0
[sh4_reg_r12] = 17
[sh4_reg_r11] = 0
[sh4_reg_r10] = 0
[sh4_reg_r9] = 61
[sh4_reg_r8] = 0
[sh4_reg_r7_bank1] = 20
[sh4_reg_r6_bank1] = 2
[sh4_reg_r5_bank1] = 4328e3ff
[sh4_reg_r4_bank1] = ffffe3ff
[sh4_reg_r3_bank1] = ce
[sh4_reg_r2_bank1] = 0
[sh4_reg_r1_bank1] = 0
[sh4_reg_r0_bank1] = 61
[sh4_reg_r7_bank0] = 0
[sh4_reg_r6_bank0] = 0
[sh4_reg_r5_bank0] = 80000060
[sh4_reg_r4_bank0] = fff
[sh4_reg_r3_bank0] = 8c149b04
[sh4_reg_r2_bank0] = 2d93f
[sh4_reg_r1_bank0] = 1492401
[sh4_reg_r0_bank0] = 80149b04
[sh4_reg_fpul] = 0
[sh4_reg_fpscr] = 0
[sh4_reg_pr] = 8c02b128
[sh4_reg_macl] = 0
[sh4_reg_mach] = 0
[sh4_reg_dbr] = 0
[sh4_reg_sgr] = 8c0b2e9c
[sh4_reg_vbr] = 8c032488
[sh4_reg_gbr] = 0
[sh4_reg_spc] = 8c0000e8
[sh4_reg_pc] = 8c032aa6
[sh4_reg_ssr] = 40000000
[sh4_reg_sr] = 70000000
[function] = "write_area0_16"
[file] = "/home/jay/programs/washingtondc/src/MemoryMap.c"
[line] = 122
[address] = 4326
[length] = 2
[feature] = "proper response for when the guest tries to write to the bios"
```

**Comment by snickerbockers** [comment]
possible root-cause: BRAF and BSRF are supposed to jump relative to the PC, but they don't add the PC to their jump destination addresses.

**Comment by snickerbockers** [comment]
nvm, they do actuallu add the PC.

**Comment by snickerbockers** [comment]
I believe that the root-cause is that my intermediate language neglects to call sh4_on_sr_change when the sr register is updated.  This also appears to be the root-cause of issue #32.

**Comment by snickerbockers** [comment]
On second-thought, issue #32 might actually be something else...

**Comment by snickerbockers** [comment]
This bug was fixed by commit e9e9425cfa9a98c2f6f4e738920da7d7546bac46.
